### ISSUE TITLE
Authenticatation error - OpenSSL: error:0909006C:PEM routines:get_name:no start line

### ISSUE THREAD

**Issue by marcpre** [googleapis/google-api-php-client#2028]
I am using `Laravel Framework 6.20.12` and `PHP 7.4.1`, with the `"google/apiclient": "^2.0"`.

OpenSSL is enabled:

[![enter image description here][1]][1]

I want to read from a spreadsheet and created the credentials on google (anonymized for obvious reasons): 

```
{
  "type": "service_account",
  "project_id": "xx",
  "private_key_id": "xxx",
  "private_key": "xxxx",
  "client_email": "xxxx",
  "client_id": "xxx",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "xxx"
}
```

I basically but the json as a string (without spaces) in an environment variable and load it the following way:

```
        $this->client = new Google_Client();
        $this->client->setApplicationName($applicationName);
        $this->client->setScopes([Google_Service_Sheets::SPREADSHEETS]);
        $this->client->setAccessType('offline');

        $jsonAuth = getenv('GOOGLE_SHEETS_API');
        $this->client->setAuthConfig(json_decode($jsonAuth, true));
		
		$sheets = new Google_Service_Sheets($this->client);

        $data = [];
        $currentRow = 2;

        // $spreadsheetId = getenv('SPREADSHEET_ID');
        $range = 'A2:E10000';
        $rows = $sheets->spreadsheets_values->get("xxxxxxxx", $range, ['majorDimension' => 'ROWS']); // <-- HERE I GET THE ERROR
```

When trying to get the spreadsheets data I get the following error:

`openssl_sign(): supplied key param cannot be coerced into a private key`

Digging into this error with `openssl_error_string()`, I get:

`error:0909006C:PEM routines:get_name:no start line`

Any suggestions what I am doing wrong?

I appreciate your replies!

  [1]: https://i.stack.imgur.com/vgMWA.png

**Comment by bshaffer** [comment]
Hi @marcpre !
What are you using for the value of `GOOGLE_SHEETS_API` (don't paste it here)? it seems that `openssl` thinks you're trying to supply a private key and the format is incorrect. Is it possible you're either using **client credentials** (e.g. you'll see `client_id` and `client_secret` in your auth config) or a PEM file (`.pem` files are the old way we used to authenticate service accounts, but that has since been updated to use `.json` files instead)
